### PR TITLE
feat(react-catalog-view-extension): Added catalog view extension and three components

### PR DESCRIPTION
**Pull request by rebeccaalpert**
Address: https://github.com/patternfly/patternfly-react/issues/3159

Heads up for @jcaianirh.

### CODE REVIEW

**Comment by rebeccaalpert** [comment]
PF4 docs build locally; not sure why the test is failing.

**Comment by rebeccaalpert** [comment]
Yes, that's the current structure.

File: packages/patternfly-4/react-catalog-view-extension/package.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{
+  "name": "@patternfly/react-catalog-view-extension",
```
Comment (jeff-phillips-18) [inline packages/patternfly-4/react-catalog-view-extension/package.json]: do we need to include the word `extension`? I think being in its own package is enough to get the point across.

Comment (dgutride) [reply packages/patternfly-4/react-catalog-view-extension/package.json]: Yes - that is the pattern we need to follow for any extension package so we can clearly understand the intent behind the package.

File: packages/patternfly-4/react-catalog-view-extension/less/catalog-item.less [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+.catalog-item-header-pf-title {
+  font-weight: 400;
+  margin-bottom: 0;
+  margin-top: 0;
+}
+
+.catalog-item-header-pf-subtitle {
+  color: @color-pf-black-500;
+  margin-bottom: 0;
+}
```
Comment (jeff-phillips-18) [inline packages/patternfly-4/react-catalog-view-extension/less/catalog-item.less]: I don't believe there is a need to support Less anymore.

Comment (rebeccaalpert) [reply packages/patternfly-4/react-catalog-view-extension/less/catalog-item.less]: I would be thrilled to remove it. I'll work on that later today.

Comment (rebeccaalpert) [reply packages/patternfly-4/react-catalog-view-extension/less/catalog-item.less]: I removed the less support.

**Comment by dlabrecq** [comment]
Can you please attach an issue?

**Comment by jeff-phillips-18** [comment]
Seem the PR title/description needs to be updated to include the rest of the catalog components.

File: packages/patternfly-4/react-catalog-view-extension/sass/patternfly-react-extensions.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+@import 'bootstrap/variables';
+@import 'bootstrap/mixins';
+
+// Patternfly Core variables and mixins
+@import 'patternfly/variables';
+@import 'patternfly/bootstrap-mixin-overrides';
+@import 'patternfly/mixins';
+
+// Extensions StyleSheets
+@import './patternfly-react-extensions/patternfly-react-extensions';
```
Comment (jeff-phillips-18) [inline packages/patternfly-4/react-catalog-view-extension/sass/patternfly-react-extensions.scss]: file name should probably be react-catalog-view-extensions.scss

File: packages/patternfly-4/react-catalog-view-extension/sass/patternfly-react-extensions/_patternfly-react-extensions.scss [hunk 4]
```diff
@@ -0,0 +1,7 @@
+/**
+  Patternfly React Extensions Partials
+*/
+@import 'variables';
+@import 'catalog-item';
+@import 'catalog-tile';
+@import 'vertical-tabs';
```
Comment (jeff-phillips-18) [inline packages/patternfly-4/react-catalog-view-extension/sass/patternfly-react-extensions/_patternfly-react-extensions.scss]: Why are we adding `VerticalTabs` to the `catalog-view`-extension package?

Comment (rebeccaalpert) [reply packages/patternfly-4/react-catalog-view-extension/sass/patternfly-react-extensions/_patternfly-react-extensions.scss]: It's a part of the original PF3 extension package and OpenShift uses it.

File: packages/patternfly-4/react-catalog-view-extension/sass/patternfly-react-extensions/_catalog-tile.scss [hunk 5]
```diff
@@ -0,0 +1,76 @@
+.ws-preview {
+  .catalog-tile-pf {
```
Comment (dlabrecq) [inline packages/patternfly-4/react-catalog-view-extension/sass/patternfly-react-extensions/_catalog-tile.scss]: Is this file for an example? I don't see the file used with an example, but noted the `ws-preview` selector we typically use there.

File: packages/patternfly-4/react-catalog-view-extension/sass/patternfly-react-extensions/_patternfly-react-extensions.scss (hunk 4 above)
Comment (jeff-phillips-18) [reply packages/patternfly-4/react-catalog-view-extension/sass/patternfly-react-extensions/_patternfly-react-extensions.scss]: but as this is not a `catalog` component, I wouldn't expect to find it in the catalog extension pacakge.

Comment (rebeccaalpert) [reply packages/patternfly-4/react-catalog-view-extension/sass/patternfly-react-extensions/_patternfly-react-extensions.scss]: @dgutride suggested the name for this; I can change it to whatever you guys agree on.

File: packages/patternfly-4/react-catalog-view-extension/src/components/CatalogTile/examples/catalogTile.scss [hunk 6]
```diff
@@ -0,0 +1,78 @@
+.ws-preview {
```
[the remark this answers is not in the document]
Comment (rebeccaalpert) [reply packages/patternfly-4/react-catalog-view-extension/src/components/CatalogTile/examples/catalogTile.scss]: Updated in subsequent commit.

File: packages/patternfly-4/react-catalog-view-extension/src/components/CatalogItemHeader/examples/catalogItemHeader.scss [hunk 7]
```diff
@@ -0,0 +1,30 @@
+.ws-preview {
```
[the remark this answers is not in the document]
Comment (rebeccaalpert) [reply packages/patternfly-4/react-catalog-view-extension/src/components/CatalogItemHeader/examples/catalogItemHeader.scss]: Updated in subsequent commit.

File: packages/patternfly-4/react-catalog-view-extension/sass/patternfly-react-extensions/_catalog-tile.scss (hunk 5 above)
Comment (rebeccaalpert) [reply packages/patternfly-4/react-catalog-view-extension/sass/patternfly-react-extensions/_catalog-tile.scss]: Good catch, I'll fix this later today.

File: packages/patternfly-4/react-catalog-view-extension/src/components/VerticalTabs/examples/verticalTab.scss [hunk 8]
```diff
@@ -0,0 +1,97 @@
+.ws-preview {
```
[the remark this answers is not in the document]
Comment (rebeccaalpert) [reply packages/patternfly-4/react-catalog-view-extension/src/components/VerticalTabs/examples/verticalTab.scss]: Updated in subsequent commit.

**Comment by redallen** [comment]
Another PR preview: https://patternfly-pr-3145.surge.sh

**Comment by jeff-phillips-18** [comment]
Will the tiles still support faded text truncation as in the PF3 extensions? 

image

**Comment by jeff-phillips-18** [comment]
The multiple icon badges seem too close together:
image

**Comment by rebeccaalpert** [comment]
It's because the CSS was getting stripped out. I'm pushing a change that should fix it. I've been working with Zack on this.

**Pull request merged**